### ISSUE TITLE
Using Portal produces a TypeScript error

### ISSUE THREAD

**Issue by aureldussauge** [chakra-ui/ark#1787]
### Description

I'm using `Portal` inside a `Popover` but it produces an error. I have to add `// @ts-ignore` to make it work

<img width="1023" alt="image" src="https://github.com/chakra-ui/ark/assets/12541829/8fb7b3dd-c77b-4dd9-9548-aef7cb8cabc8">

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/p/devbox/hardcore-frog-kfkk9c?file=%2Fapp%2Fpage.tsx%3A16%2C29

### Steps to Reproduce

1. Use `Portal` component in a React App

### Ark UI Version

1.1.0

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

**Comment by cschroeter** [comment]
@aureldussauge 

Thanks for taking your time to create this issue. This is apparently a Codesandbox only problem since I could not reproduce this in a Next.js, Astro or component library setup but Codesandbox.

**Comment by cschroeter** [comment]
@aureldussauge 

I've took a closer look and apparently TypeScript resolved the types to:

```ts
export declare const Portal: (props: PropsWithChildren<PortalProps>) => ReactNode;
```

Which is not correct. So I've added ean xplicit return type so the result looks like this:

```ts
export declare const Portal: (props: PropsWithChildren<PortalProps>) => ReactPortal[] | ReactNode | null | undefined;
```

I'm not sure if that helps but since its anyway just a Codesandbox thing I will close this issue. The "possible" fix for this will be availabe in the next release.
